src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -337,216 +334,73 @@ public static <K, V extends TopLevelItem> Map<K, V> loadChildren(AbstractFolder<
+                saveIfNeeded(item, itemNeedsSave);
                 configurations.put(key.apply(item), item);
             } catch (Exception e) {
-                Logger.getLogger(ItemGroupMixIn.class.getName()).log(Level.WARNING, "could not load " + subdir, e);
+                File finalEffectiveSubdir = effectiveSubdir;
+                LOGGER.log(Level.WARNING, e, () -> "could not load " + finalEffectiveSubdir);
             }
         }
 
         return configurations;
     }
 
-    @Override
-    public String getItemName(File dir, I item) {
-        ChildNameGenerator<AbstractFolder<I>, I> childNameGenerator = childNameGenerator();
-        String itemName;
-        if (childNameGenerator == null) {
-            itemName = dir.getName();
-        } else {
-            File nameFile = new File(dir, ChildNameGenerator.CHILD_NAME_FILE);
-            if (nameFile.isFile()) {
-                try {
-                    itemName = StringUtils.trimToNull(Files.readString(nameFile.toPath(), StandardCharsets.UTF_8));
-                } catch (IOException e) {
-                    LOGGER.log(Level.WARNING, e, () ->"Caught an IOException while trying to read " + nameFile + ", assuming child name is " + dir.getName());
-                    itemName = null;
-                }
-                if (itemName == null) {
-                    LOGGER.log(Level.WARNING, "{0} was empty, assuming child name is {1}", new Object[]{nameFile, dir.getName()});
-                    itemName = dir.getName();
-                }
-            } else {
-                itemName = dir.getName();
+    private static <V extends TopLevelItem> void saveIfNeeded(V item, boolean itemNeedsSave) {
+        if (itemNeedsSave) {
+            try {
+                item.save();
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Could not update {0} after applying folder naming rules",
+                        item.getFullName());
             }
         }
-        String finalItemName = itemName;
-        LOGGER.log(Level.FINE, () -> "itemName = " + finalItemName);
-        return itemName;
+    }
+
+    @Override
+    public String getItemName(File dir, I item) {
+        return childNameGenerator().readItemName(dir);
     }
 
     protected final I itemsPut(String name, I item) {
-        ChildNameGenerator<AbstractFolder<I>, I> childNameGenerator = childNameGenerator();
-        if (childNameGenerator != null) {
-            File nameFile = new File(getRootDirFor(item), ChildNameGenerator.CHILD_NAME_FILE);
-            String oldName;
-            if (nameFile.isFile()) {
-                try {
-                    oldName = StringUtils.trimToNull(Files.readString(nameFile.toPath(), StandardCharsets.UTF_8));
-                } catch (IOException e) {
-                    oldName = null;
-                }
-            } else {
-                oldName = null;
-            }
-            if (!name.equals(oldName)) {
```

Comment:
@Vlatombe Before this PR, we only attempted to write the child name file when the new name was not equal to the old name. After this PR, we started unconditionally writing the child name file, which at the very minimum introduced a performance regression (updating the file modification time unnecessarily) and may have introduced a worse regression if other consumers tried to read the file while it was being (unnecessarily) modified.

Reply:
Re-reading the code, the logic just moved to https://github.com/jenkinsci/cloudbees-folder-plugin/blob/5338bc4d6992a1b0791360f7793fbb650aeffaca/src/main/java/com/cloudbees/hudson/plugins/folder/ChildNameGenerator.java#L327-L339. I think the behaviour is the same as previously.

Reply:
@vlatombe Only after I fixed the regression in https://github.com/jenkinsci/cloudbees-folder-plugin/pull/482.

Reply:
Ah, right. Thank you for it!